docs/workflows/reconnection-logic.md

```diff
@@ -0,0 +1,91 @@
+
+#### 2. The client stops sending data. 
+
+This scenario is a bit more tricky. The application is still alive, however the OS may decide to not run the game or send any data while the application is in the background. [SpatialOS]({{urlRoot}}/reference/glossary#spatialos-runtime) will close the connection, if it doesn't receive any messages from a client-worker for a period of time. You need add a way to reconnect clients to your game to handle this scenario. 
+
+## Types of disconnections in the GDK
+
+[Heartbeating]({{urlRoot}}/modules/player-lifecycle/heartbeating) is a technique used to verify that your client is still connected. If there are too many failed heartbeats, there are two ways a client-worker may be seen as disconnected:
+
+### 1. Heartbeating fails in the SpatialOS Runtime
+
+To ensure that a worker is still connected, the worker has to send messages to SpatialOS at a set interval. This indicates that the worker is still alive. If SpatialOS doesn't receive any messages for a while, it will close the connection to that worker. 
+
+Whenever the connection gets closed on a worker, that worker receives a `DisconnectOp` object. It contains the reason behind the disconnection and triggers a disconnect event in the GDK.
+
+#### How to handle a disconnect
+
+The GDK provides you with a callback that is triggered when the worker receives a `DisconnectOp` object. You can register to this callback to perform your disconnection logic.
+
+```csharp
+namespace YourGame
+{
+    public class ClientWorkerConnector : WorkerConnector
+    {
+        ...
+
+        protected override void HandleWorkerConnectionEstablished()
+        {
+             Worker.OnDisconnect += OnDisconnected;
+        }
+
+        private void OnDisconnected(string reason)
+        {
+            Worker.LogDispatcher.HandleLog(LogType.Log, new LogEvent($"Worker disconnected")
+                .WithField("WorkerId", Worker.WorkerId)
+                .WithField("Reason", reason));
+        }
+    }
+}
+```
+
+
+
+### 2. Heartbeating fails in the player lifecycle module 
+
+Our [Player Lifecycle module]({{urlRoot}}/modules/player-lifecycle/overview) allows you to easily create players and uses the heartbeating technique to detect and delete the player entity of any unresponsive client-workers. In mobile applications, this can result in entities being destroyed even though the worker is still connected to SpatialOS. If this happens, that client-worker won't receive a `DisconnectOp`.
+
+#### How to handle a disconnect
+
+When the player entity gets deleted, the client-worker might end up with no entities to be authoritative over and therefore unable to check out any entities. This leaves the client-worker in a bad state. The safest option to handle this is to disconnect at this point and go back to your start screen to attempt a reconnect.
+
+```csharp
+namespace YourGame
+{
+    public class YourMonoBehaviour : MonoBehaviour
```

Comment:
Maybe note that this MonoBehaviour would on your player GameObject?

Reply:
```suggestion
    // This MoonBehaviour needs to be placed on your player prefab.
    public class YourMonoBehaviour : MonoBehaviour
```

?

Reply:
```MoonBehaviour```

😂

Reply:
🌚